sonar-plugin-api/src/main/java/org/sonar/api/server/profile/BuiltInQualityProfilesDefinition.java

```diff
@@ -0,0 +1,419 @@
+
+    List<BuiltInActiveRule> rules();
+  }
+
+  @Immutable
+  class BuiltInQualityProfileImpl implements BuiltInQualityProfile {
+
+    private static final Logger LOG = Loggers.get(BuiltInQualityProfilesDefinition.BuiltInQualityProfileImpl.class);
+    private final String language;
+    private final String name;
+    private final boolean isDefault;
+    private final Map<RuleKey, BuiltInActiveRule> activeRulesByKey;
+
+    private BuiltInQualityProfileImpl(NewBuiltInQualityProfileImpl newProfile) {
+      this.name = newProfile.name();
+      this.language = newProfile.language();
+      this.isDefault = newProfile.isDefault();
+
+      Map<RuleKey, BuiltInActiveRule> ruleBuilder = new HashMap<>();
+      for (NewBuiltInActiveRule newActiveRule : newProfile.activeRules()) {
+        ruleBuilder.put(RuleKey.of(newActiveRule.repoKey, newActiveRule.ruleKey), new BuiltInActiveRule(newActiveRule));
+      }
+      this.activeRulesByKey = unmodifiableMap(ruleBuilder);
+    }
+
+    @Override
+    public String language() {
+      return language;
+    }
+
+    @Override
+    public String name() {
+      return name;
+    }
+
+    @Override
+    public boolean isDefault() {
+      return isDefault;
+    }
+
+    @Override
+    @CheckForNull
+    public BuiltInActiveRule rule(RuleKey ruleKey) {
+      return activeRulesByKey.get(ruleKey);
+    }
+
+    @Override
+    public List<BuiltInActiveRule> rules() {
+      return unmodifiableList(new ArrayList<>(activeRulesByKey.values()));
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      BuiltInQualityProfileImpl that = (BuiltInQualityProfileImpl) o;
+      return language.equals(that.language) && name.equals(that.name);
+    }
+
+    @Override
+    public int hashCode() {
+      int result = name.hashCode();
+      result = 31 * result + language.hashCode();
+      return result;
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder("BuiltInQualityProfile{");
+      sb.append("name='").append(name).append('\'');
+      sb.append(", language='").append(language).append('\'');
+      sb.append(", default='").append(isDefault).append('\'');
+      sb.append('}');
+      return sb.toString();
+    }
+  }
+
+  class NewBuiltInActiveRule {
+    private final String repoKey;
+    private final String ruleKey;
+    private String overridenSeverity = null;
+    private final Map<String, NewOverridenParam> paramsByKey = new HashMap<>();
+
+    private NewBuiltInActiveRule(String repoKey, String ruleKey) {
+      this.repoKey = repoKey;
+      this.ruleKey = ruleKey;
+    }
+
+    public String repoKey() {
+      return this.repoKey;
+    }
+
+    public String ruleKey() {
+      return this.ruleKey;
+    }
+
+    public NewBuiltInActiveRule overrideSeverity(String s) {
```

Comment:
Javadoc should link to `org.sonar.api.rule.Severity` constants. It should also mention that the default rule severity is used by default.